### ISSUE TITLE
Fix broken keyboard navigation in Web Modeler (a11y)

### ISSUE THREAD

**Issue by lmbateman** [camunda/camunda-modeler#4628]
### Describe the bug

1. Skip links are broken throughout the Modeler.
2. I can put focus on "Modeler" and "Home" in the header, but nothing happens when I press Return.
3. I can't seem to tab around inside the Help Center at all. Pressing Tab with the Help Center open just moves me around the page behind the Help Center.
4. Some focus indicators in the Collaborators panel and the Collaborators modal are missing or too subtle (see attached screenshots).
5. The Action button in tables doesn't have a focus indicator.
6. In the canvas, the filename and overflow button don't have focus indicators.
7. I can't use the keyboard to navigate through the overflow menu.
8. After the overflow button, the tab sequence becomes completely opaque. Focus seems to move through the canvas tools, then becomes visible at the AI button. Then it moves to the context pad, eventually reaches the Details panel, and then it skips over the palette and bottom panel and starts over. I'm not sure how much of this is design issues and how much is due to the canvas being SVG.
9. The focus indicator on the Share button in the DMN editor is too subtle.

### Steps to reproduce

Open the Modeler and navigate through each page with the keyboard.

### Expected behavior

- I should be able to skip to the main content of each page using the skip link.
- I should be able to access every interactive element on the page using the keyboard.
- Non-interactive elements should not receive focus.
- When a modal is open, focus should be trapped inside it. 
- Focus should always be visible.

### Environment

- OS: MacOS 14.7
- Camunda Modeler Version: Web
- Execution Platform: Camunda 8

### Additional context

Broken skip link: 
![image](https://github.com/user-attachments/assets/dbdc6c67-b43f-4818-ad45-d73cebbdb83b)

Focus where nothing happens:
![image](https://github.com/user-attachments/assets/5c0a01f4-6938-46de-bead-d519697b1b74)
![image](https://github.com/user-attachments/assets/0a05b516-bb17-47c0-ba6c-c03a6c8c9a07)

Broken tabbing in Help Center:
![image](https://github.com/user-attachments/assets/bd01df01-9bc5-40d8-85ab-ee124b116787)

Missing/too-subtle focus indicators in Collaborators UI:
![image](https://github.com/user-attachments/assets/2448397d-f2e1-46b6-9e37-8919809886b4)
![image](https://github.com/user-attachments/assets/e602f478-0ac1-48ea-b4ec-f1fa98615c6e)

Missing focus indicator on Action buttons:
![image](https://github.com/user-attachments/assets/ec3ee2dc-ab4e-424b-8e12-72deaba9b73b)

Missing focus indicator for filename and overflow button:
![image](https://github.com/user-attachments/assets/3a447374-ea4f-49c4-b4af-4be1d5c31179)

Too-subtle focus indicator on Share button in DMN editor:
![image](https://github.com/user-attachments/assets/7ff2909d-a3b1-41cb-aef8-a5b2e599b938)
